### ISSUE TITLE
[GeoMechanicsApplication] Extract the non-linear geometry functionality (Updated Lagrange) from elements

### ISSUE THREAD

**Issue by rfaasse** [KratosMultiphysics/Kratos#12058]
As a developer, I want to extract the non-linear geometry functionality from the elements, such that the element classes are more maintainable and configurable, by only depending on the abstract interface of a newly created non-linear geometry policy.

### Acceptance criteria
- All functionality related to non-linear geometries, which could have different implementations (in our case Updated Lagrange, but could be extended) is moved to a separate class using the strategy design pattern see e.g. [wikipedia](https://en.wikipedia.org/wiki/Strategy_pattern) for a bit more info
- The interface of the policy is passed to an element, such that the implementation is hidden.
- All updated lagrangian classes are retained until a better registration mechanism is thought out, but their only function should be creating a version of their base with the correct policy.

**_Note: For more details, see also the discussion in #12057_** 

We need to understand how important this feature is. Also: what is the cost of delaying this?

**Comment by rfaasse** [comment]
This needs to be crystallized out regarding the:
1. Extra `CalculateOnIntegrationPoints` functionality
2. Geometric stiffness matrix (KG)

To be taken up by @avdg81 @WPK4FEM

**Comment by indigocoral** [comment]
@rfaasse , @avdg81 , @WPK4FEM Is this still needed?

**Comment by WPK4FEM** [comment]
> [@rfaasse](https://github.com/rfaasse) , [@avdg81](https://github.com/avdg81) , [@WPK4FEM](https://github.com/WPK4FEM) Is this still needed?

My short reply: yes.
It will reduce code duplication and eventually reduce the number of element classes that we need to maintain.

**Comment by avdg81** [comment]
I agree with @WPK4FEM. However, do we expect to work on this any time in 2026? If no, do we still want to keep this issue open? Or should we close it with a comment and open a new one when the need for nonlinear geometric functionality arises? The latter will contribute to a cleaner product backlog.
